velox/connectors/tpch/TpchConnector.cpp

```diff
@@ -0,0 +1,144 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/tpch/TpchConnector.h"
+#include "velox/tpch/gen/TpchGen.h"
+
+namespace facebook::velox::connector::tpch {
+
+using facebook::velox::tpch::Table;
+
+namespace {
+
+RowVectorPtr getTpchData(
+    Table table,
+    size_t maxRows,
+    size_t offset,
+    size_t scaleFactor,
+    memory::MemoryPool* pool) {
+  switch (table) {
+    case Table::TBL_PART:
+      return velox::tpch::genTpchPart(maxRows, offset, scaleFactor, pool);
+    case Table::TBL_SUPPLIER:
+      return velox::tpch::genTpchSupplier(maxRows, offset, scaleFactor, pool);
+    case Table::TBL_PARTSUPP:
+      return velox::tpch::genTpchPartSupp(maxRows, offset, scaleFactor, pool);
+    case Table::TBL_CUSTOMER:
+      return velox::tpch::genTpchCustomer(maxRows, offset, scaleFactor, pool);
+    case Table::TBL_ORDERS:
+      return velox::tpch::genTpchOrders(maxRows, offset, scaleFactor, pool);
+    case Table::TBL_LINEITEM:
+      return velox::tpch::genTpchLineItem(maxRows, offset, scaleFactor, pool);
+    case Table::TBL_NATION:
+      return velox::tpch::genTpchNation(maxRows, offset, scaleFactor, pool);
+    case Table::TBL_REGION:
+      return velox::tpch::genTpchRegion(maxRows, offset, scaleFactor, pool);
+  }
+  return nullptr;
+}
+
+} // namespace
+
+TpchDataSource::TpchDataSource(
+    const std::shared_ptr<const RowType>& outputType,
+    const std::shared_ptr<connector::ConnectorTableHandle>& tableHandle,
+    const std::unordered_map<
+        std::string,
+        std::shared_ptr<connector::ColumnHandle>>& columnHandles,
+    velox::memory::MemoryPool* FOLLY_NONNULL pool)
+    : pool_(pool) {
+  auto tpchTableHandle =
+      std::dynamic_pointer_cast<TpchTableHandle>(tableHandle);
+  VELOX_CHECK_NOT_NULL(
+      tpchTableHandle, "TableHandle must be an instance of TpchTableHandle");
+  tpchTable_ = tpchTableHandle->getTable();
+
+  auto tpchTableSchema = getTableSchema(tpchTableHandle->getTable());
+  VELOX_CHECK_NOT_NULL(tpchTableSchema, "TpchSchema can't be null.");
+
+  outputColumnMappings_.reserve(outputType->size());
+
+  for (const auto& outputName : outputType->names()) {
+    auto it = columnHandles.find(outputName);
+    VELOX_CHECK(
+        it != columnHandles.end(),
+        "ColumnHandle is missing for output column: '{}'",
+        outputName);
+
+    auto handle = std::dynamic_pointer_cast<TpchColumnHandle>(it->second);
+    VELOX_CHECK_NOT_NULL(
+        handle,
+        "ColumnHandle must be an instance of TpchColumnHandle for '{}'",
```

Comment:
Please add the table and column name to these assertion messages for easier debugging.